(SPRINT) Migration, Auth, Profile, Layouts

VulpoTheDev opened the pull request:
# THIS PR IS A SPRINT
This pull request is a combination of pull request

## Sprint Big Picture

<img width="573" alt="image" src="https://github.com/user-attachments/assets/3cdee71e-d9bb-4203-8b36-fa0a8861568f" />

## Requirements for merge

- [x] Has to merge (all merge conflicts must be resolved)
- [x] Project has successfully migrated to Next.js version 15
- [x] Authentication with ability to sign in, register, use oauth, forget password and restricted pages
- [ ] Profile page must be completed and match design
- [x] Layouts (Navbar, Sidebar, Footer) and their components ie: Notification must be completed and match design
- [ ] https://github.com/MyArtverse-Project/MyArtverse/pull/66#discussion_r2009831459
- [ ] Must pass all tests and be able to be deployed to vercel
- [ ] Approval from @VulpoTheDev & @kuroji-fusky

VulpoTheDev committed:
chore: login page

VulpoTheDev committed:
chore: work in progress login page

VulpoTheDev committed:
chore: login server actions

VulpoTheDev committed:
buiild(deps): bump nextjs 15

VulpoTheDev committed:
chore: login flow

VulpoTheDev committed:
Next15 migration (#61)

VulpoTheDev commented:
Issues whilist migrating to v15

<img width="1114" alt="image" src="https://github.com/user-attachments/assets/32778c5a-7c3c-42d8-b83e-ff94c1982dd1" />
<img width="1600" alt="image" src="https://github.com/user-attachments/assets/b5bde1d4-8a36-4406-b4db-4670695ec5be" />

kuroji-fusky commented:
> Issues whilist migrating to v15
> 
> <img alt="image" width="1114" src="https://private-user-images.githubusercontent.com/86149507/420777006-32778c5a-7c3c-42d8-b83e-ff94c1982dd1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2NzgxMjMsIm5iZiI6MTc0MTY3NzgyMywicGF0aCI6Ii84NjE0OTUwNy80MjA3NzcwMDYtMzI3NzhjNWEtN2MzYy00MmQ4LWI4M2UtZmY5NGMxOTgyZGQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzExVDA3MjM0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc5Y2I2N2M4MDdiYWQ4ZmY0ZjlkNzVhZDI1OTllYTYwNzM1MTI5ZjJhZGNhNzUyNDUwZWFmY2YwMjdjN2JmZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jqDvvfN-DKLC_PSaROV0eYBKaKHBPy0jueVSSI7PTOU"> <img alt="image" width="1600" src="https://private-user-images.githubusercontent.com/86149507/420777083-b5bde1d4-8a36-4406-b4db-4670695ec5be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2NzgxMjMsIm5iZiI6MTc0MTY3NzgyMywicGF0aCI6Ii84NjE0OTUwNy80MjA3NzcwODMtYjViZGUxZDQtOGEzNi00NDA2LWI0ZGItNDY3MDY5NWVjNWJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzExVDA3MjM0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0M2FjM2IyMDU5OTRjZjgwOGIzNjk3ZmZlMmM5OGFhYmNlYmZhNThhYjVkZDdmZjQwY2VkOGUyODNmNTBmYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VBmKbxrWipw0-Mr0fcLnFwMssMndSGqSjRzeIvAr65U">

Sooo it turns out React 19 deprecated the `forwardRef` on most of the input and button components - nice, will be doing a major refactor on those said components

<https://react.dev/blog/2024/12/05/react-19#ref-as-a-prop>

kuroji-fusky committed:
refactor deprecated `forwardRef` components

kuroji-fusky committed:
minor ui changes to `/login`

kuroji-fusky committed:
minor: use `cn` util to merge tw classes

kuroji-fusky committed:
cleanup: use custom ref type

kuroji-fusky committed:
minor prettier changes

VulpoTheDev committed:
chore: Improved Login Page design for Ultrawide montiors

VulpoTheDev committed:
chore: Registration & Verify Page + Logic

VulpoTheDev commented:
https://github.com/MyArtverse-Project/MyArtverse/blob/da7ed0219971661b33e0bd663994ba18b522fc93/packages/ui/components/comments/CommentBase.tsx#L9 goes through an issue
```
./src/app/(profile)/profile/page.tsx:30:17
Type error: Type 'ForwardRefExoticComponent<Omit<DetailedHTMLProps<ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>, "height" | ... 5 more ... | "srcSet"> & { ...; } & RefAttributes<...>>' is not assignable to type 'ForwardRefExoticComponent<any>'.
  Type 'React.ReactNode' is not assignable to type 'import("/Users/jasondiazjimenez/Documents/GitHub/MyArtverse/node_modules/@types/react-dom/node_modules/@types/react/index").ReactNode'.
    Type 'bigint' is not assignable to type 'ReactNode'.

  28 |             <div className="mb-5">
  29 |               <UserCommentInput
> 30 |                 imgTag={Image}
     |                 ^
  31 |                 avatar="/img/examples/kuro/kuro-example4.png"
  32 |               />
  33 |             </div>
Next.js build worker exited with code: 1 and signal: null
```

VulpoTheDev committed:
deps(react): updated react

VulpoTheDev committed:
chore: replace useValidateEmail for zod

VulpoTheDev committed:
fix(verify): use promises on params

VulpoTheDev committed:
chore: workarounds for vercel to deploy

VulpoTheDev committed:
chore(eslintrc): warn on conisitent type imports

VulpoTheDev committed:
chore: changes for eslint and turbo to deploy

kuroji-fusky committed:
refactor: use `React` type namespace

kuroji-fusky committed:
chore(actions): separate linting and unused code checks

chore(actions): remove `--prefer-offline`
chore(actions): ensure only `push` events are triggered only

VulpoTheDev committed:
chore: forgot password page + logic

VulpoTheDev committed:
fix(recover): use promises on params

kuroji-fusky committed:
chore: update GitHub Actions workflow for improved formatting and linting checks

kuroji-fusky committed:
refactor: move validate children hook to its own hook dir

kuroji-fusky committed:
fix(types): coerced types on `useValidateChildrenComponents` hook

This is refactored without running the dev server, so I'm frankly hoping that this error will go away

VulpoTheDev committed:
chore: Oauth buttons goes to backend for login

VulpoTheDev committed:
Ignored issues for the time being with TODO

VulpoTheDev committed:
chore: moved profile to [handle]

VulpoTheDev committed:
chore: updated typings

VulpoTheDev committed:
chore: Sidebar

VulpoTheDev committed:
chore: Navbar Actions

VulpoTheDev committed:
feat: Theme toggle in site settings

VulpoTheDev committed:
fix(dropdownItem): theme toggle padding

VulpoTheDev committed:
chore: gradient for request artist access

VulpoTheDev committed:
chore: updated footer to match design

VulpoTheDev committed:
chore: added compliance pages and refresh token

VulpoTheDev committed:
chore: profile masthead

VulpoTheDev committed:
fix: merge issues

VulpoTheDev committed:
chore: api fetchers

kuroji-fusky committed:
chore: added helper type for async routes

code copied from https://github.com/kuroji-fusky/SponsorExplorer/blob/29e120a/client/src/types/routeParams.ts#L3-L17

kuroji-fusky committed:
fix(typo): variant typo

VulpoTheDev committed:
chore: character profile page

VulpoTheDev committed:
chore: added modals

VulpoTheDev committed:
chore: search bar and create folder modal

VulpoTheDev committed:
chore: Integrated Folders with API

kuroji-fusky committed:
chore: use `AsyncProps`

kuroji-fusky committed:
chore: updated logo

- updated footer links with semantic HTML

VulpoTheDev committed:
chore: WIP Character Masthead

VulpoTheDev committed:
chore: add some spacing to tabs

VulpoTheDev committed:
feat: added html editor modal

VulpoTheDev committed:
feat: customizable panels

VulpoTheDev committed:
chore: added images

VulpoTheDev committed:
chore: display panels

VulpoTheDev committed:
feat: comments

VulpoTheDev committed:
chore: use asyncprops on params

kuroji-fusky committed:
chore: prettier format

- set `trailingComma` to `none`
- remove redudant imports

kuroji-fusky committed:
chore: refactored functions and types

- use the `cn` utility for `ThemeToggle
- refactored `displayer.ts`
- fixed deprecated type from `React.MutableRefObject`
- removed redudant `aria-disabled` attribute on `Checkbox.tsx` - the `disabled` attribute already includes this

VulpoTheDev committed:
chore: dynamic panel loading

kuroji-fusky committed:
chore(deps): update `next` version

just a quick dep update due to a critical vulnerability - we should be fine since vercel takes care of everything

https://zhero-web-sec.github.io/research-and-things/nextjs-and-the-corrupt-middleware

VulpoTheDev committed:
feat: post comments, info panel

kuroji-fusky committed:
fix(minor): fixed typo on "nationality"

kuroji-fusky committed:
chore: distinguish type imports

kuroji-fusky committed:
minor: remove unused file

`mergeNamespaceComponents` is dormant and won't be used again

kuroji-fusky committed:
chore: housekeeping and formatting

kuroji-fusky committed:
chore: migrate ESLint and Prettier configs to Biome

kuroji-fusky committed:
minor: changed lint commands to biome

-  tweaking some biome settings

I didn't run any commands to prevent a merge conflict

kuroji-fusky committed:
refactor(actions): modified commands to biome

kuroji-fusky committed:
chore(deps): removed eslint and prettier

kuroji-fusky committed:
chore: update .vercelignore to ignore desktop deployments

kuroji-fusky committed:
fix(minor): fix speling for `silent`

VulpoTheDev committed:
chore: replies on comments

VulpoTheDev committed:
Merge branch 'character' of https://github.com/MyArtverse-Project/MyArtverse into character

kuroji-fusky committed:
fix(imports): remove unused `mergeNamespaceComponents` import

VulpoTheDev committed:
chore: view replies function

VulpoTheDev committed:
chore: relationship modals

VulpoTheDev committed:
chore: ability to view profile logged out

kuroji-fusky committed:
Migrate from ESLint and Prettier to Biome (#67)

Removes both ESLint and Prettier's config files and dependencies; migrates all the configs from Biome

VulpoTheDev committed:
chore: improved profile visibility

VulpoTheDev committed:
Merge branch 'character' of https://github.com/MyArtverse-Project/MyArtverse into character

VulpoTheDev committed:
format: biome

VulpoTheDev committed:
chore: return null on silent errors

VulpoTheDev committed:
chore: display notifications

VulpoTheDev committed:
feat: search button

VulpoTheDev committed:
chore: wip search

VulpoTheDev committed:
chore: clickable links

VulpoTheDev committed:
chore: Character (#66)

VulpoTheDev committed:
feat: Layout (#64)

VulpoTheDev committed:
chore: housecleaning (#62)

Currently the project is unable to deploy to vercel, in this PR I made
changes that involves the UI repository, before putting it in #60 I want
to make sure that all changes made will not break the functionality of
the website in the long run

VulpoTheDev committed:
format(biome): formatted code

kuroji-fusky commented on apps/web/.eslintrc.json:
What's this doing here? Did you run a Next migrate script or something? We're already formatting and linting our code with Biome already

VulpoTheDev replied:
That was on me I was on a different branch

kuroji-fusky replied:
Ah I see, looks like I'll migrate from the API repo as well lol

kuroji-fusky commented on apps/web/src/app/(main)/(settings)/settings/(account)/profile/page.tsx:
```diff
@@ -16,8 +16,8 @@ export default function SettingsProfilePage() {
-            standards. If you are certain that you'll delete your account, it's
-            important to export your data first before proceeding.
+            standards. If you are certain that you{"'"}ll delete your account,
+            it{"'"}s important to export your data first before proceeding.
```
This looks awkward - this is probably something to do with the ESLint rule of React's unescape characters or something; it's nothing major to revert this as it's more of a cosmetic change, and there are no viz changes when viewed through browsers, I'd suggest reverting it and disabling the said rule

```suggestion
            standards. If you are certain that you'll delete your account, it's
            important to export your data first before proceeding.
```

kuroji-fusky commented on apps/web/src/components/Modals/CreateFolder.tsx:
```diff
@@ -24,7 +24,7 @@ export default function CreateFolderModal({
 }) {
   const [folderName, setFolderName] = React.useState<string>("")
-  const [color, setColor] = React.useState<string>("")
+  const [color, _setColor] = React.useState<string>("")
```
Using `React.useState` seems really redundant here, maybe use the `useState` instead of importing the entire React lib?

kuroji-fusky commented on apps/web/src/components/layouts/Cards/Status.tsx:
```diff
@@ -1,11 +1,11 @@
+import type { FursonaStatus } from "@/types/characters"
```
Since we're targeting to a wide range of audience, consider renaming the "fursona" imports?

```suggestion
import type { CharacterStatus } from "@/types/characters"
```

kuroji-fusky commented on package.json:
```diff
@@ -13,7 +13,7 @@
     "format": "biome format",
     "format:write": "biome format --write",
     "lint": "biome lint",
```
Must've made an oversight on the lint command, there is no such thing as a "lint" command in Biome:

```suggestion
    "lint": "biome check",
```

kuroji-fusky commented on apps/web/src/components/Search/SearchBar.tsx:
```diff
@@ -1,19 +1,29 @@
+    artwork: [],
+    character: []
+  })
   const [query, setQuery] = useState("")
```
This would be more clearer, because it sounds like a backend operation

```suggestion
  const [searchQuery, setSearchQuery] = useState("")
```

kuroji-fusky reviewed (commented):
There's a lot of substantial changes so far, but it looks like there's some more housekeeping to do

kuroji-fusky commented on apps/web/public/Backdrop.png:
This should be exported as an SVG, since they support gradients, and we use blur on a `<div>` container

kuroji-fusky commented on apps/web/src/components/Search/SearchBar.tsx:
```diff
@@ -0,0 +1,161 @@
+                isCharacter
+              />
+            )}
+            {!loading &&
+              query.trim() &&
+              results.user?.length === 0 &&
+              results.artwork?.length === 0 &&
+              results.character?.length === 0 && (
```
This is too complex to read and comprehend, consider simplifying the logic and when we add a result type (i.e. a listing), we don't need to chain another logic like `... && results.listing?.length === 0`

```suggestion
  const hasNoSearchResults =
      query.trim() &&
      Object.values(results).some((v) => v.length === 0)
      
   return (
            {/* ... */}
            {!loading && hasNoSearchResults (
```

kuroji-fusky commented on apps/web/src/components/Search/Section.tsx:
```diff
@@ -0,0 +1,73 @@
+}
+
+export function SearchSection({ title, items }: SearchSectionProps) {
+  let formattedItems: { name: string; image?: string; href: string }[] = []
+
+  if (!items) {
+    return null
+  }
+
+  if (Array.isArray(items)) {
+    formattedItems = items.map((item) =>
+      typeof item === "string"
+        ? { name: item }
+        : {
+            name: item.name,
+            image: (item as { image?: string }).image,
+            href: `/search?q=${item.name}`
+          }
+    )
+  } else if (typeof items === "object") {
+    formattedItems = [
+      ...(items.user?.map((user: UserType) => ({
+        name: user.displayName || user.handle,
+        image: user.avatarUrl,
+        href: `/@${user.handle}`
+      })) ?? []),
+      ...(items.character?.map((char: Character) => ({
+        name: char.name,
+        image: char.avatarUrl,
+        href: `/@${char.owner.handle}/${char.name}`
+      })) ?? []),
+      ...(items.artwork?.map((art: Artwork) => ({
+        name: art.title || "Untitled Artwork",
+        image: art.artworkUrl || art.watermarkUrl,
+        href: `/artworks/${art.id}`
+      })) ?? [])
+    ]
+  }
```
This one's strange, maybe `useState` or `useReducer` can help with this?

kuroji-fusky reviewed (changes requested):
Another housekeeping work:

- The backdrop should be converted to SVG; the image is half a megabyte
- Search components need to use hooks instead of raw dogging with `let` on [src/components/Search/Section.tsx](https://github.com/MyArtverse-Project/MyArtverse/pull/60/files#diff-8027711c35e859f402a5df8c15386b16ecb3a7588b75555116621a930fd10c09)
- Most of the components from `apps/web` need to be moved from `packages/ui` soon, so the desktop app will get the same treatment

kuroji-fusky committed:
chore(refactor): simplify complex search logic

- removes eslint
- renamed `query` to `searchQuery`

resolves:
- https://github.com/MyArtverse-Project/MyArtverse/pull/60#discussion_r2032206796

kuroji-fusky committed:
chore(fix): correct biome "lint" command

resolves: https://github.com/MyArtverse-Project/MyArtverse/pull/60#discussion_r2032192999

kuroji-fusky committed:
chore(refactor): import `useState` directly

resolves: https://github.com/MyArtverse-Project/MyArtverse/pull/60#discussion_r2032185532

kuroji-fusky committed:
chore(fix): unwrap apostrophe characters

resolves: https://github.com/MyArtverse-Project/MyArtverse/pull/60#discussion_r2032183108

kuroji-fusky committed:
chore(refactor): replaced "fursonas" with "characters"

- used star imports
- extracted props as their own interfaces

kuroji-fusky committed:
refactor(minor): remove eslint comments

kuroji-fusky committed:
chore(refactor): use star imports

kuroji-fusky committed:
minor(refactor): add a delimiter

VulpoTheDev committed:
fix: removeSuffix for auth forms
